front/lib/api/assistant/jit_actions.ts

```diff
@@ -340,6 +334,50 @@ export async function renderConversationForModelJIT({
     }
   }
 
+  if (selected.length > 0) {
+    const { filesAsXML, hasFiles } = await listConversationFiles({
+      conversation,
+    });
+    if (hasFiles) {
+      const randomCallId =
+        "list_conversation_files_" + Math.random().toString(36).substring(7);
+      const functionName = "list_conversation_files";
+
+      const simulatedAgentMessages = [
+        // 1. We add a message from the agent, asking to use the files listing function
+        {
+          role: "assistant",
+          content: "Please list the files available to this conversation.",
```

Comment:
I don't think we need any content here. Also this specific content is a little strange, as this would be the model talking (if it generates content along with the fcalls, it would look more like "I'm going to use tool `list_conversation_files` in order to obtain the list of conversation files first"

Reply:
I was hesitating as, beyond the name, we don't have a description of what the function does.
I was trying to hint the model.

Do you think I should update the content or get rid of it altogether ?

Reply:
> we don't have a description of what the function does.
I was trying to hint the model.

Fair point.

Personally I would remove it (function name is fairly explicit already) because it may clash with the user's prompt (eg "use <thinking> before using a tool" or "always provide an evaluation of whether the tool was useful after using a tool" etc..).

If we're keeping it (I'm not strongly against it), I would change it to be more chain-of-thought-y, because the current version doesn't sound like something a model would generate before using a tool (kind of like if you said out loud "please push this commit onto the remote." as you are typing "git push" in your terminal :p)

Reply:
I'll remove it and see how it goes.